### PR TITLE
Multi docker stack

### PR DESCRIPTION
**Pull request by Dieterbe**
refactor into several docker-compose stacks
    
    variants:
    * simple standalone with CI built docker image and binary, monitors into itself
    * similar, but mounts custom built binary, monitors into itself
    * clustered stack with kafka input and graphite for monitoring
    
    also:
    - remove unneeded links
    - re-arrange a bit
    - export cassandra port so you can query it from the host if you want
    - simplify hostnames (strip -1 everywhere)
    
    Notably missing from this work, but available in
    https://github.com/raintank/raintank-docker is tooling to automatically
    run given workload scripts. We can add that later, it's not a priority now .

### CODE REVIEW

**Comment by Dieterbe** [comment]
note i have run the docker-standard stack and it seems to work fine, just like the previous docker stack

File: docker/docker-cluster/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    ports:
+     - "8125:8125/udp"
+    volumes:
+     - "./statsdaemon.ini:/etc/statsdaemon.ini"
+
+  # easiest to just use a separate graphite server for monitoring
+  graphitemon:
+    hostname: graphitemon
+    image:  raintank/graphite
+    ports:
```
Comment (woodsaj) [inline docker/docker-cluster/docker-compose.yml]: this image does not exist.  We were building it as part of raintank-docker.
Perhaps we should use https://hub.docker.com/r/raintank/graphite-stack/  for both graphite statsd and grafana.

**Comment by Dieterbe** [comment]
pushed a commit that uses graphite-stack. the problem is however that it only speaks https.

i would like the stacks to be as consistent/similar as possible, and only differ where needed. e.g. ideally the populate.sh script should work for all stacks, meaning the same datasource config should work for all stacks. but in the other docker stacks we use graphite-api+metrictank for monitoring which only accepts plain http (https://github.com/raintank/graphite-metrictank/blob/master/pkg/Dockerfile)
for docker-cluster i can route port 8000 to 443 but the requests will fail since it will refuse plain http requests.

**Comment by woodsaj** [comment]
only graphite-web uses https, which you dont need to use.  Grafana uses http on port 3000

**Comment by Dieterbe** [comment]
> So why cant you just change the datasource using 8000 use port 443 and https?

then the script would create a DS that is broken for the dev and standard stacks.
I was trying to see if we could have the same populate.sh script for all stacks, which means the same DS configs, and we would use the different docker-compose.ymls to provide different implementations for the graphite datasource. This would work if graphite-stack would serve graphite over http, which it doesn't.

But anyway it's not worth the time trying to go that route, I'll just set up different datasources based on which environment you're in.

**Comment by Dieterbe** [comment]
@woodsaj how does it look now?

note:
* the monitoring in cluster stack using graphite-stack doesn't work because the SSL certificate doesn't verify
* the extra/populate.sh script is not documented yet. the docker guide still asks people to set things up manually which I think is a good exercise. also i'm not certain yet I like the workflow and how it takes an argument. perhaps it should be run from inside the specific docker stack dir

**Pull request merged**